### DISCUSSION TITLE
Marker label does not allow scanning of the values if X-Axis scrolling is consumed

### DISCUSSION THREAD

**Discussion by Josuhu** [patrykandpatrick/vico#355] [Ideas]
There is great potential for this chart library and therefore I bring this user experience issue forward.

Marker label does not allow scanning of the values if X-Axis scrolling is consumed. Generally for the user it is important to stop scrolling and being able to scan through the values what he sees on the screen.

Should there be a condition programmed where the marker label scan has priority over the scrolling consuming if user long presses the chart e.g. for 1s and does not actually scroll left or right. Then again if the user immediately swipes to scroll the scroll consume has the priority over the marker label scanning?

you can repeat the current behavior with all charts at the demo app.

**Comment by Gowsky** [comment]
@Josuhu, thank you for your suggestion. It may be a nice feature to have.
However, in our experience mixing various gestures (like long tap and swipe in your example) causes lots of issues and is nearly impossible to make right in Jetpack Compose. I'm not worried about implementing this feature in good old `View` system. But due to the fact that we have to keep features as close as possible in both systems, we will not be implementing it anytime soon. Though we are open for contributions, or samples of you have this feature semi-working.

**Comment by Josuhu** [comment]
Thanks for the info on your perspective and good to know it is under consideration. I really like this library and maybe I also contribute some code examples if the chart example source code contains everything I need to study more of this feature... Could you please point me the correct class paths to look more from the source code?

**Reply by patrickmichalik** [reply]
Hello! `Charts.kt` (`compose`) and `BaseChartView.kt` (`views`) should be good starting points.

**Reply by Josuhu** [reply]
Here is one simple code example and a video demo for dedicated scan field solution. The Vico library is advanced and takes more time to get familiar for more elegant solutions built in. With this at least developer could activate the function via a Chart parameter if offered as option. Users tend to adapt in various functions via trial and error even if they are hidden as this would be.

https://github.com/patrykandpatrick/vico/assets/55630343/920af24a-a9ce-48c0-b9a8-fc1f56867b24

In this demo at the Carts.kt class I just added another "dedicated touch" element at bottom and on top of the Canvas element. For the sake of demo it has red borders.
![image](https://github.com/patrykandpatrick/vico/assets/55630343/e4ab4fc1-ad0f-4ac6-80a0-61c729e6e06d)

**Reply by patrickmichalik** [reply]
Thanks for the idea, @Josuhu. We’d like to make this feature accessible in charts with no `HorizontalAxis` too, so we believe your initial suggestion is optimal. We’ll look into implementing such a mechanism.

**Comment by y9san9** [comment]
I am also looking forward this issue. My users report current behaviour as a bug. They can scan values in one direction (like when you are at the end of scroll), but can't in the other (in this case scroll performed)

**Reply by y9san9** [reply]
@patrickmichalik, Is there a possibility that you can add an ability to scroll using two fingers, and one finger will be used to scan values?

**Reply by patrickmichalik** [reply]
We’d like to avoid implementing nonstandard behavior, @y9san9. Scrolling is typically tied to one-finger swipes. However, we’ll be looking into adding the mechanism that @Josuhu initially described.

**Comment by Tyler-Lopez** [comment]
Hi there, just curious: have there been any updates on this front?

**Reply by Gowsky** [reply]
When we have updates to share, we’ll post them here. We want to look into this and other `CartesianMarker`-related improvements relatively soon.
